modules/c/nrt/unittests/test_byte_swap.c

```diff
@@ -0,0 +1,70 @@
+/* =========================================================================
+ * This file is part of NITRO
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2016, MDA Information Systems LLC
+ *
+ * NITRO is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program;
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <import/nrt.h>
+#include "Test.h"
+
+
+TEST_CASE(testFourBytes)
+{
+    nrt_Uint32 value;
+    nrt_Uint32 swappedValue;
+
+    value = 0x03040506;
```

Comment:
To make sure there's nothing weird happening with upper bits for some reason, can you change this to something like `0x12345678`?